### PR TITLE
proper fix for migrating profiles and same name profiles.

### PR DESCRIPTION
**Pull request by DevBefell**

### CODE REVIEW

File: launcher-patches/0027-proper-fix-for-migrating-profiles-and-same-name-prof.patch [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+-import com.google.gson.GsonBuilder;
+-import com.google.gson.JsonDeserializationContext;
+-import com.google.gson.JsonDeserializer;
+-import com.google.gson.JsonElement;
+-import com.google.gson.JsonObject;
+-import com.google.gson.JsonParseException;
+-import com.google.gson.JsonParser;
+-import com.google.gson.JsonSerializationContext;
+-import com.google.gson.JsonSerializer;
++import com.google.gson.*;
```
Comment (bigfoot547) [inline launcher-patches/0027-proper-fix-for-migrating-profiles-and-same-name-prof.patch:126]: Could you please reconfigure your IDE to not prefer on-demand imports? I don't want MORE conflicts due to import shuffling.

File: launcher-patches/0027-proper-fix-for-migrating-profiles-and-same-name-prof.patch [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+ import net.minecraft.launcher.Launcher;
+ import net.minecraft.launcher.LauncherConstants;
+@@ -37,6 +15,11 @@ import org.apache.commons.io.FileUtils;
+ import org.apache.logging.log4j.LogManager;
+ import org.apache.logging.log4j.Logger;
+ 
++import java.io.File;
++import java.io.IOException;
++import java.lang.reflect.Type;
++import java.util.*;
```
Comment (bigfoot547) [inline launcher-patches/0027-proper-fix-for-migrating-profiles-and-same-name-prof.patch:154]: ditto

File: launcher-patches/0027-proper-fix-for-migrating-profiles-and-same-name-prof.patch [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+ import java.awt.event.ActionEvent;
+ import java.awt.event.ActionListener;
+ import java.awt.event.MouseAdapter;
+ import java.awt.event.MouseEvent;
+ import java.io.IOException;
+-import java.util.ArrayList;
+-import java.util.Collection;
+-import java.util.Collections;
+-import java.util.List;
++import java.util.*;
```
Comment (bigfoot547) [inline launcher-patches/0027-proper-fix-for-migrating-profiles-and-same-name-prof.patch:342]: ditto

**Review by bigfoot547 — CHANGES_REQUESTED** [review]
just the imports and stuff (trying to phase on-demand imports out of this project)

thanks for contributing

**Comment by DevBefell** [comment]
fixed the commit.

**Review by bigfoot547 — APPROVED** [review]
looks good to me and works

**Pull request merged**